src/app/decorators/captcha-decorator.ts

```diff
@@ -0,0 +1,9 @@
+import {ValidationOptions} from "./ValidationOptions";
+import { MetadataStorageAndValidation } from './MetadataStorage';
+import { getFromContainer } from './container';
+
+export function IsCaptcha(validationOptions?: ValidationOptions) {
```

Comment:
I think we can simplify this by creating a custom validation class that implements ValidatorConstraintInterface, this is documented in the [README](https://github.com/typestack/class-validator/blob/master/README.md). You already have the validation function figured out but I think we can accomplish it without having to duplicate the files in our project.